### PR TITLE
taint source id for bit-cuda

### PR DESCRIPTION
**Pull request by ZcortezZ**
added field `taint_sources` on BitCudaTransformer that stores the id of nodes containing a taint source

### CODE REVIEW

File: src/bit_cuda/bit_cuda_transformer.h [hunk 1]
```diff
@@ -50,42 +55,50 @@ class BitCudaTransformer : public CfgVisitor<LatticeType>
...
 
         if(node.arguments.size() >= 1){
             node_struct.first_transfer_index = add_transfer_function(node.formal_parameters[0], node.arguments[0]);
+            add_id(id, transfer_functions[node_struct.first_transfer_index].rhs);
             bit_cuda::Transfer& last = transfer_functions[node_struct.first_transfer_index];
             
             for (int i = 1; i < node.arguments.size(); i++)
             {
                 last.next_transfer_index = add_transfer_function(node.formal_parameters[i], node.arguments[i]);
+                add_id(id, transfer_functions[last.next_transfer_index].rhs);
```
Comment (thorulf4) [inline src/bit_cuda/bit_cuda_transformer.h:72]: If the taint constant is used in two seperate arguments and thus get a add_id call. Wouldnt the node id be added to taint_sources twice?

**Review by thorulf4 — CHANGES_REQUESTED** [review]

File: src/bit_cuda/bit_cuda_transformer.h [hunk 2]
```diff
@@ -133,6 +147,15 @@ class BitCudaTransformer : public CfgVisitor<LatticeType>
             }
         } 
     }
+    void add_id(int bitcudaId, int transfer[]){
```
Comment (NicklasHugoy) [inline src/bit_cuda/bit_cuda_transformer.h:150]: Maybe `add_taint_sources` is a more descriptive name

Comment (NicklasHugoy) [reply src/bit_cuda/bit_cuda_transformer.h:150]: and `bitcudaId` -> `nodeIndex`

**Review by NicklasHugoy — APPROVED** [review]

File: src/bit_cuda/bit_cuda_transformer.h (hunk 1 above)
Comment (ZcortezZ) [reply src/bit_cuda/bit_cuda_transformer.h]: changed to store it in a set instead of a vector

File: src/bit_cuda/bit_cuda_transformer.h (hunk 2 above)
Comment (ZcortezZ) [reply src/bit_cuda/bit_cuda_transformer.h]: renamed them

**Review by thorulf4 — APPROVED** [review]

**Review by NicklasHugoy — APPROVED** [review]

**Pull request merged**